app/src/main/java/com/github/orkest/ui/feed/FeedActivity.kt

```diff
@@ -240,11 +249,54 @@ private fun PlayButton(song: Song){
 }
 
 @Composable
-private fun Reaction(post: Post){
+private fun LikeButton(viewModel: PostViewModel, post: Post) {
+
+    //Declaring variables to update the UI
+    val isPostLiked = remember{ mutableStateOf(false) }
+    val nbLikes = remember { mutableStateOf(post.nbLikes) }
+
+    //Fetching initial values from database
+    viewModel.isPostLiked(post).thenApply { isPostLiked.value = it }
+    viewModel.getNbLikes(post).thenApply { nbLikes.value = it }
+
+    val buttonColor = if (isPostLiked.value) Color.Yellow else Color.White
+
+    Column {
+        //Displaying the like button
+        Icon(
+            painter = painterResource(id = R.drawable.black_like_icon),
+            contentDescription = "Like button",
+            tint = buttonColor,
+            modifier = Modifier
+                .testTag("like_button")
+                .height(20.dp)
+                .width(20.dp)
+                .clickable {
+                    viewModel.updatePostLikes(post, !isPostLiked.value).thenApply {
+                        //Updating the isPostLiked value accordingly
+                        isPostLiked.value = !isPostLiked.value
+                        //Updating the value of getNbLikes only after the updatePostLikes future has been completed
+                        viewModel.getNbLikes(post).thenApply { nbLikes.value = it }
+                    }
+                },
+        )
+        //Displaying the number of likes for this post
+        Text(
+            text = "${nbLikes.value}",
```

Comment:
you can directly apply nbLikes.value.toString(), there is no need for formatting as it is the only displayed value